plugins/services/src/js/containers/services/ServicesTable.js

```diff
@@ -651,4 +657,28 @@ ServicesTable.propTypes = {
   services: PropTypes.array
 };
 
-module.exports = ServicesTable;
+// TODO: move into it's own file
+function withMasterRegionName(Component) {
+  const master$ = MesosStateStore.getMasterRequestStream();
+  const masterRegionName$ = master$.pipe(
+    map(masterRequestString => {
+      const masterRequest = JSON.parse(masterRequestString);
+      const nodeRegionName = findNestedPropertyInObject(
+        masterRequest,
+        "get_master.master_info.domain.fault_domain.region.name"
+      );
+
+      return nodeRegionName || "N/A";
```

Comment:
Maybe its as simple as making this component pure, but should we also go ahead and establish the pattern for making sure this component only re-renders when the region.name changed and no on every stream update?